### ISSUE TITLE
Progress bar does not present training loss step when Ranger optimizer is used.

### ISSUE THREAD

**Issue by psandev** [PyTorchLightning/pytorch-lightning#6749]
## 🐛 Bug
AdamW:
![adam](https://user-images.githubusercontent.com/30928228/113021736-8ce1f680-918c-11eb-86fd-76bd3270c04e.png)
Ranger:
![ranger](https://user-images.githubusercontent.com/30928228/113021792-9bc8a900-918c-11eb-8531-6e3f2725c5bb.png)

### To Reproduce

Use pytorch_ranger or Less Wright's package https://github.com/lessw2020/Ranger-Deep-Learning-Optimizer 

The code is in my repo:
https://github.com/psandev/PyTorch_Lightning_YOLOv3

### Environment

**Note**: `Bugs with code` are solved faster ! `Colab Notebook` should be made `public` !

* `IDE`: Please, use our python [bug_report_model.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py
) template.

* `Colab Notebook`: Please copy and paste the output from our [environment collection script](https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py) (or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```

 - PyTorch Version (1.71):
 - OS (Linux):
 - How you installed PyTorch (pip):
 - Build command you used (NA):
 - Python version: 3.7
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration:
 - Any other relevant information:

**Comment by akihironitta** [comment]
@psandev I wasn't able to reproduce the bug with the following script using Ranger optimizer.

<details><summary>code</summary>

```python
import os

import torch
from torch.utils.data import Dataset

from pl_examples import cli_lightning_logo
from pytorch_lightning import LightningModule, Trainer


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def loss(self, batch, prediction):
        # An arbitrary loss to have a loss that updates the model weights during `Trainer.fit` calls
        return torch.nn.functional.mse_loss(prediction, torch.ones_like(prediction))

    def step(self, x):
        x = self.layer(x)
        out = torch.nn.functional.mse_loss(x, torch.ones_like(x))
        return out

    def training_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        return {"loss": loss}

    def training_step_end(self, training_step_outputs):
        return training_step_outputs

    def training_epoch_end(self, outputs) -> None:
        torch.stack([x["loss"] for x in outputs]).mean()

    def validation_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        return {"x": loss}

    def validation_epoch_end(self, outputs) -> None:
        torch.stack([x['x'] for x in outputs]).mean()

    def test_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        return {"y": loss}

    def test_epoch_end(self, outputs) -> None:
        torch.stack([x["y"] for x in outputs]).mean()

    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.layer.parameters(), lr=0.1)
        lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
        return [optimizer], [lr_scheduler]


def test_run():
    import time
    from ranger import Ranger

    class TestModel(BoringModel):
        def configure_optimizers(self):
            return Ranger(self.parameters(), lr=1e-4, weight_decay=1e-5, eps=1e-9)

        def on_train_batch_end(self, outputs, batch, batch_idx, dataloader_idx):
            time.sleep(1)

    # fake data
    train_data = torch.utils.data.DataLoader(RandomDataset(32, 64))
    val_data = torch.utils.data.DataLoader(RandomDataset(32, 64))
    test_data = torch.utils.data.DataLoader(RandomDataset(32, 64))

    # model
    model = TestModel()
    trainer = Trainer(
        default_root_dir=os.getcwd(),
        limit_train_batches=10,
        limit_val_batches=1,
        max_epochs=10,
        weights_summary=None,
    )
    trainer.fit(model, train_data, val_data)
    # trainer.test(test_dataloaders=test_data)


if __name__ == '__main__':
    test_run()
```

</details>

<details><summary>env</summary>

```console
$ python collect_env_details.py
* CUDA:
	- GPU:
	- available:         False
	- version:           10.2
* Packages:
	- numpy:             1.20.1
	- pyTorch_debug:     False
	- pyTorch_version:   1.8.1+cu102
	- pytorch-lightning: 1.3.0dev
	- tqdm:              4.56.2
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         
	- python:            3.8.5
	- version:           #1 SMP Debian 4.19.171-2 (2021-01-30)
```

</details>

Could you provide a minimal script using the [BoringModel](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py)?
